transport-native-epoll/src/main/c/netty_epoll_linuxsocket.c

```diff
@@ -262,6 +267,134 @@ static jobject netty_epoll_linuxsocket_getPeerCredentials(JNIEnv *env, jclass cl
      (*env)->SetIntArrayRegion(env, gids, 0, 1, (jint*) &credentials.gid);
      return (*env)->NewObject(env, peerCredentialsClass, peerCredentialsMethodId, credentials.pid, credentials.uid, gids);
 }
+
+static jint netty_epoll_linuxsocket_openTunTapCloneDevice0(JNIEnv* env, jclass clazz, jstring cloneDevName) {
+    const char* f_cloneDevName = (*env)->GetStringUTFChars(env, cloneDevName, 0);
+
+    // Open specific clone device.  Raise an IOException if an error occurs.
+    int res = open(f_cloneDevName, O_RDWR|O_NONBLOCK|O_CLOEXEC);
+    if (res < 0) {
+        netty_unix_errors_throwChannelExceptionErrorNo(env, "openTunTapCloneDevice() failed: ", errno);
+    }
+
+    (*env)->ReleaseStringUTFChars(env, cloneDevName, f_cloneDevName);
+
+    return res;
+}
+
+static jint netty_epoll_linuxsocket_allocTunTapInterface(JNIEnv *env, jclass cls, jint cloneDevFD, jstring ifName, jboolean isTapDev)
+{
+    jint res;
+    struct ifreq ifr;
+    const char *f_ifName = (*env)->GetStringUTFChars(env, ifName, NULL);
+
+    // Initialize the IFREQ structure with the specified flags and interface name.
+    memset(&ifr, 0, sizeof(ifr));
+    ifr.ifr_flags = ((isTapDev) ? IFF_TAP : IFF_TUN);
+    strncpy(ifr.ifr_name, f_ifName, IFNAMSIZ);
```

Comment:
do we need to pad the remaining value with `0`s, we just memset it to 0 above. If we just care that `ifr.ifr_name` is null terminated consider using `strcat`.